mdx-web/src/main/java/com/mx/path/model/mdx/web/controller/AuthenticationController.java

```diff
@@ -29,8 +32,66 @@
 @RequestMapping(value = "{clientId}", produces = BaseController.MDX_MEDIA)
 public class AuthenticationController extends BaseController {
 
-  @RequestMapping(value = { "/authentications", "/sessions" }, method = RequestMethod.POST, consumes = { MDX_MEDIA, MDX_ONDEMAND_MEDIA }, produces = { MDX_MEDIA, MDX_ONDEMAND_MEDIA })
-  public final ResponseEntity<Authentication> authenticate(@PathVariable("clientId") String clientId, @RequestBody Authentication requestAuthentication) {
+  @SuppressWarnings("MagicNumber")
+  @RequestMapping(value = "/authentications", method = RequestMethod.POST, consumes = MDX_MEDIA)
+  public final ResponseEntity<?> authenticate(@PathVariable("clientId") String clientId, HttpServletRequest request) {
+    return versioned(request)
+        .defaultVersion(Authentication.class, Authentication.class, (authentication -> {
+          return authenticateLegacy(clientId, authentication);
+        }))
+        .version(20240209, Authentication20240209.class, Authentication20240209.class, (authentication -> {
+          return authenticate(clientId, authentication);
+        }))
+        .execute();
+  }
+
+  @RequestMapping(value = { "/sessions" }, method = RequestMethod.POST, consumes = MDX_ONDEMAND_MEDIA, produces = MDX_ONDEMAND_MEDIA)
```

Comment:
Will OnDemand authentication always use the latest API version? Will this require any changes in Grunt?

Reply:
It will stay on v5 unless we update the specs for MDX OnDemand. We don't own that spec, so we need to continue to support v5. That's why I separated it.